lib/index.js

```diff
@@ -37,10 +49,14 @@ class GraphQLComponent {
 
     this._imports = [];
 
+    this._dataSource = dataSource;
```

Comment:
what if a component wants to use more than 1 data source without doing composition with another component? We already have components in the monorepo that utilize more than 1 service-client (ie. they talk to more than 1 service). Are we going to force each component to implement a datasource that encapsulates more than 1 data source? I figured a DataSource class should likely be 1 to 1 with an underlying data source. That way DataSources could be re-used across components if necessary.

Reply:
Sure, makes sense. Added.

Reply:
Although, the only problem here is that `component.dataSources` will now return an array which means you won't know what the data source actually is.

I guess the right thing to do is probably change it to a map.

Reply:
And that leads to another problem, because the point of having a single dataSource relationship and `name` property on the data source is that there is less guessing at what namespace was used.

Reply:
I'll take a look at your changes but yeah I think you captured what I was thinking. The constructor takes an array of DataSource objects (who each have a name) and then converts that to a map of datasource objects keyed by name for access in resolvers.